### PR TITLE
Add new registry architechture interfaces

### PR DESCRIPTION
**Pull request by gpsanant**

### CODE REVIEW

File: src/contracts/interfaces/IRegistryCoordinator.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+/**
+ * @title Interface for a contract the coordinates between various registries for an AVS.
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IRegistryCoordinator.sol:5]: typo: 'the' => 'that'

File: src/contracts/interfaces/IRegistryCoordinator.sol [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    enum Status
+    {
+        // default is inactive
+        INACTIVE,
+        ACTIVE
+    }
+
+    /**
+     * @notice  Data structure for storing info on operators to be used for:
+     * - sending data by the sequencer
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IRegistryCoordinator.sol:19]: EigenDA-specific comment?

File: src/contracts/interfaces/IRegistryCoordinator.sol [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    function operatorIdToQuorumBitmap(bytes32 pubkeyHash) external view returns (uint256);
+
+    /// @notice Returns the stored id for the specified `operator`.
+    function getOperatorId(address operator) external view returns (bytes32);
+
+    /// @notice Returns task number from when `operator` has been registered.
+    function getFromTaskNumberForOperator(address operator) external view returns (uint32);
+
+    /// @notice Returns the registry at the desired index
+    function registries(uint256) external view returns (address);
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IRegistryCoordinator.sol:41]: probably want a function for the length of this list as well?

File: src/contracts/interfaces/IStakeRegistry.sol [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+ */
+interface IStakeRegistry is IRegistry {
+    // DATA STRUCTURES
+
+    /// @notice struct used to store the stakes of an individual operator or the sum of all operators' stakes, for storage
+    struct OperatorStakeUpdate {
+        // the block number at which the stake amounts were updated and stored
+        uint32 updateBlockNumber;
+        // the block number at which the *next update* occurred.
+        /// @notice This entry has the value **0** until another update takes place.
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IStakeRegistry.sol:18]: just commenting that it seems like there's a good property or two to check here

File: src/contracts/interfaces/IStakeRegistry.sol [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        uint32 nextUpdateBlockNumber;
+        // stake weight for the quorum
+        uint96 stake;
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IStakeRegistry.sol:21]: maybe this should be called `weight` given its description?

File: src/contracts/interfaces/IStakeRegistry.sol [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    /**
+     * @notice Registers the `operator` with `operatorId` for the specified `quorumNumbers`.
+     * @param operator The address of the operator to register.
+     * @param operatorId The id of the operator to register.
+     * @param quorumNumbers The quorum numbers the operator is registering for.
+     * @dev Permissioned by RegistryCoordinator
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IStakeRegistry.sol:29]: IMO this is kind of confusing wording. I would suggest 'permissioned to' or 'access restricted to the'

File: src/contracts/interfaces/IStakeRegistry.sol [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+     */
+    function getStakeUpdateForQuorumFromOperatorIdAndIndex(uint8 quorumNumber, bytes32 operatorId, uint256 index)
+        external
+        view
+        returns (OperatorStakeUpdate memory);
+
+    /**
+     * @notice Returns the stake weight corresponding to `operatorId` for quorum `quorumNumber`, at the
+     * `index`-th entry in the `operatorIdToStakeHistory[operatorId][quorumNumber]` array if it was the operator's
+     * stake at `blockNumber`. Reverts otherwise.
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IStakeRegistry.sol:64]: I think the wording "if it was the operator's stake at `blockNumber`" should be improved

File: src/contracts/interfaces/IStakeRegistry.sol [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+     * @param quorumNumber The quorum number to get the stake for.
+     * @param operatorId The id of the operator of interest.
+     * @param index Array index for lookup, within the dynamic array `operatorIdToStakeHistory[operatorId][quorumNumber]`.
+     * @param blockNumber Block number to make sure the stake is from.
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IStakeRegistry.sol:68]: perhaps this is a good place to clarify better on the exact expected usage of this input

Comment (ChaoticWalrus) [reply src/contracts/interfaces/IStakeRegistry.sol:68]: same issue is present in `getTotalStakeAtBlockNumberFromIndex`

File: src/contracts/interfaces/IRegistryCoordinator.sol [hunk 9]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+/**
+ * @title Interface for a contract the coordinates between various registries for an AVS.
+ * @author Layr Labs, Inc.
+ */
+interface IRegistryCoordinator {
```
Comment (ChaoticWalrus) [inline src/contracts/interfaces/IRegistryCoordinator.sol:8]: I still prefer the 'subregistry' language, but perhaps can be settled later

**Comment by ChaoticWalrus** [comment]
CI is rather broken, but my comments are all just nitpicks.
This is a PR to an experimental branch so merge away when you want

File: src/contracts/interfaces/IStakeRegistry.sol (hunk 5 above)
Comment (gpsanant) [reply src/contracts/interfaces/IStakeRegistry.sol:21]: i want to stick with stake for this, imo has the lindy. weight probably will confuse a bit

File: src/contracts/interfaces/IStakeRegistry.sol (hunk 4 above)
Comment (gpsanant) [reply src/contracts/interfaces/IStakeRegistry.sol:18]: agree

File: src/contracts/interfaces/IRegistryCoordinator.sol (hunk 9 above)
Comment (gpsanant) [reply src/contracts/interfaces/IRegistryCoordinator.sol:8]: got it

File: src/contracts/interfaces/IStakeRegistry.sol (hunk 8 above)
Comment (gpsanant) [reply src/contracts/interfaces/IStakeRegistry.sol:68]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/7a220382b4fdbe6880cbec7103c3e5db6a036f24

File: src/contracts/interfaces/IStakeRegistry.sol (hunk 7 above)
Comment (gpsanant) [reply src/contracts/interfaces/IStakeRegistry.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/7a220382b4fdbe6880cbec7103c3e5db6a036f24

File: src/contracts/interfaces/IStakeRegistry.sol (hunk 6 above)
Comment (gpsanant) [reply src/contracts/interfaces/IStakeRegistry.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/7a220382b4fdbe6880cbec7103c3e5db6a036f24

File: src/contracts/interfaces/IRegistryCoordinator.sol (hunk 3 above)
Comment (gpsanant) [reply src/contracts/interfaces/IRegistryCoordinator.sol:41]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/7a220382b4fdbe6880cbec7103c3e5db6a036f24

File: src/contracts/interfaces/IRegistryCoordinator.sol (hunk 2 above)
Comment (gpsanant) [reply src/contracts/interfaces/IRegistryCoordinator.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/2580b31c45e16badb4b3870a6ee7b25c93514470

File: src/contracts/interfaces/IRegistryCoordinator.sol (hunk 1 above)
Comment (gpsanant) [reply src/contracts/interfaces/IRegistryCoordinator.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/2580b31c45e16badb4b3870a6ee7b25c93514470

**Pull request merged**